tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java

```diff
@@ -1851,6 +1851,60 @@ public String transform(String source) {
                     return source;
                 }
             });
+
+            functionMap.put("personNameOrder", new Transform<String, String>() {
+                @Override
+                public String transform(String source) {
+                    // The various personName attribute values in desired sort order
+                    final List<String> lengthValues = Arrays.asList("long", "medium", "short", "monogram", "monogramNarrow");
+                    final List<String> usageValues = Arrays.asList("addressing", "referring", "sorting");
+                    final List<String> styleValues = Arrays.asList("formal", "informal");
+                    final List<String> orderValues = Arrays.asList("surnameFirst", "givenFirst");
+
+                    List<String> parts = HYPHEN_SPLITTER.splitToList(source);
+                    order = 0;
+                    for (String part: parts) {
+                        if (lengthValues.contains(part)) {
+                            order += lengthValues.indexOf(part) * 64;
```

Comment:
I think this needs to be refined.
We might get collisions. 
long = 0 would be zero, and collide with addressing = 0.
Also medium + short (1 + 2) = monogram (3).

Reply:
What I suggest instead is having one List with all the values. When you get an index, add (1 << index) to the order.

Reply:
Oh, good idea, I will do that; I forgot we could have multiple instances of the same type.